tcms/testcases/views.py

```diff
@@ -830,15 +825,31 @@ def clone(request, template_name='testcases/clone.html'):
                                  messages.SUCCESS,
                                  _('TestCase cloning was successful'))
             return HttpResponseRedirect(reverse('plans-search'))
-    else:
+
+    def get(self, request):
+        if not self._is_request_data_valid(request):
+            return HttpResponseRedirect(request.META.get('HTTP_REFERER', '/'))
+
         selected_cases = get_selected_testcases(request)
         # Initial the clone case form
         clone_form = CloneCaseForm(initial={
             'case': selected_cases,
         })
         clone_form.populate(case_ids=selected_cases)
 
-    context = {
-        'form': clone_form,
-    }
-    return render(request, template_name, context)
+        context = {
+            'form': clone_form,
+        }
+        return render(request, self.template_name, context)
+
+    def _is_request_data_valid(self, request):
+        request_data = getattr(request, request.method)
+
+        if 'case' not in request_data:
+            messages.add_message(request,
+                                 messages.ERROR,
+                                 _('At least one TestCase is required'))
+            # redirect back where we came from
```

Comment:
I don't think we need this comment. The purpose of this method is to return `True` if the request is valid and `False` if not - no redirects are involved. Maybe it was valid before, but not anymore

This is a good example of how comments in the code tend to rot and become misleading at some point, and should be avoided as much as possible

Reply:
Code will always tell you what it is doing. Comments should tell you why you need to do it.